src/main/java/org/test/device/controller/DeviceController.java

```diff
@@ -0,0 +1,32 @@
+package org.test.device.controller;
+
+import org.test.device.domain.Device;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import javax.validation.Valid;
+
+/**
+ * Created by Alireza on 1/6/2019.
+ */
+@Controller("/api")
+public class DeviceController {
+
+
+    @PostMapping(name = "/devices")
+    public ResponseEntity devices(@RequestBody @Valid Device request, Errors errors) {
+        try {
+            if (!errors.hasErrors()) {
+                return new ResponseEntity(HttpStatus.CREATED);
+            } else {
+                return new ResponseEntity(HttpStatus.BAD_REQUEST);
+            }
+        } catch (Throwable e) {
```

Comment:
Catching a Throwable in application code is generally a bad idea because you'll be catching errors such as OutOfMemoryError that your code cannot recover from. It's better to stick to Exception unless there is a recent recommendation from Spring to do so that I don't know of.

Reply:
```suggestion
        } catch (Exception e) {
```
Would it be OK?